packages/core/src/utils/user.ts

```diff
@@ -62,3 +65,33 @@ export const transpileUserProfileResponse = (
     ...(ssoIdentities && { ssoIdentities }),
   };
 };
+
+// Not used yet, may be used in the future, keep as a individual method.
+const getValidPhoneNumber = (phone: string): string => {
+  if (!phone) {
+    return phone;
+  }
+
+  try {
+    const phoneNumber = parseE164PhoneNumberWithError(phone);
```

Comment:
Better use the parsePhoneNumber function from @logto/shared as the SSOT, see my updates [here](https://github.com/logto-io/logto/pull/7374). So you may let the function throws an error if the number is invalid.